### PR TITLE
[Tight deadline] Adding new spell (DAO sector) that allows to view the DAO addresses and DAO transactions. [Ready for merging]

### PR DESCRIPTION
**Pull request by henrystats**
The following DAO creator tools were selected to be included in this spell: 

* Aragon
* DAO Haus
* Colony
* Zodiac (reality.eth)
* Syndicate (coming soon)

Brief comments on the purpose of your changes:

*For Dune Engine V2*
I've checked that:
General checks:
* [x] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [x] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [x] if adding a new model, I added a test
* [x] the filename is unique and ends with .sql
* [x] each sql file is a select statement and has only one view, table or function defined  
* [x] column names are `lowercase_snake_cased`
* [x] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [x] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributors)

Join logic:
* [x] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

Incremental logic:
* [x] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
  * [x] where block_time >= date_trunc("day", now() - interval '1 week')
* [x] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [x] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

**Comment by augustog** [comment]
Hello @henrystats, thanks for the contribution!

There's a few issues on this PR that are breaking pre-merge checks, could you please review them?
- Seems like you have duplicates, I couldn't find any obvious source of dupes so it might be good to investigate the cause, we have a known case of duplication (being fixed by # ) so if you find any of those documents in your dataset I'd wait until it's merged
- There's a few issues with metadata (ymls), a couple of fields have no descriptions and a few refs have no schema defined, few pointers that may help:
     - You can reuse column definitions across the schema (and sources) yml files by following the pattern you see in this [file](https://github.com/duneanalytics/spellbook/blob/main/models/uniswap/ethereum/uniswap_ethereum_schema.yml)
     -  Seems like you're missing schema files for many queries, you can consolidate them all in one file and use the above logic to DRY (you can see that done in the [uniswap trades spell](https://github.com/duneanalytics/spellbook/tree/main/models/uniswap/ethereum))
- Since the logic is somewhat long I would like to see at least a few tests cases for each type of DAO.

Please let us know if you need any support with any of this, we're always happy to help.

**Comment by augustog** [comment]
cc @soispoke, tagging you here since you were working on another spell on the DAO sector, I doubt there's a lot of overlap but just to bring this to your radar

(no input needed)

**Comment by soispoke** [comment]
@augustog Yeah I saw this, great timing thanks for flagging 👍

**Comment by henrystats** [comment]
Hola @augustog , thanks for the feedback, will get to work asap. 

- For the duplicates, I guess creating new schema files will help for that. Although I've tested them out and noticed the dupes for one of the creator tools, and when I checked the transaction on etherscan, there were actually multiple instances of the same token & same amount being transferred multiple times in the same transaction. How could I possibly sort this?

This was the test query I wrote - <https://dune.com/queries/1420156> 

Here's an example of a dupe: <0x8f5dafa1cf8a7dde654488396a08178a8d23a4c2b630cbda05deb686d57ad14c>

- I'll update the schema files, I thought I could avoid creating a schema since I didn't plan on exposing the spells, but I'll update that and see if the tests continue to fail.

**Comment by henrystats** [comment]
Also, I don't understand what you mean here - <Since the logic is somewhat long I would like to see at least a few tests cases for each type of DAO.>

**Comment by henrystats** [comment]
Also, the part of my schema's without description are the gnosis sources that I added to the base sources. The Base source previously did not have description for some columns hence why I skipped them since I copied what was in the ethereum base sources.

**Comment by henrystats** [comment]
dupes & schema now fixed! ty ser!

**Comment by Hosuke** [comment]
Hi @henrystats , it is an impressive work.
As there are many type of DAO models, we can add a seed file for testing like we did in [`dex.trades`](https://github.com/duneanalytics/spellbook/blob/main/seeds/dex/trades/dex_trades_seed.csv), and specify the test in the schema for every spell we added.

**Comment by henrystats** [comment]
Thank you ser @Hosuke, pushing an update tonight and will include the tests.

**Comment by henrystats** [comment]
@augustog @Hosuke I can't get the tests to work, I can't find the error defined in gh action. 

I don't have any of the characters in my schema columns, so not sure what is causing this error.

File: seeds/daos/daos_addresses_seed.csv
Comment (Hosuke) [inline seeds/daos/daos_addresses_seed.csv:1]: I think the [test failure](https://github.com/duneanalytics/spellbook/actions/runs/3354183001/jobs/5557561271) may be caused by the blank ` `  in the ending of this line.
```suggestion
blockchain,dao_creator_tool,dao,dao_wallet_address,created_date
```

Comment (henrystats) [reply seeds/daos/daos_addresses_seed.csv]: omg, how did you find that? you're legend ser!

**Review by Hosuke — CHANGES_REQUESTED** [review]
You may need to write up something similar as [dex_trade_token_bought](https://github.com/duneanalytics/spellbook/blob/main/tests/generic/dex_trades_token_bought.sql) test.

Or maybe you can try out the ideas in this [PR](https://github.com/duneanalytics/spellbook/pull/1707) (But this PR is still under review.)

File: models/daos/transactions/ethereum/daos_transactions_ethereum_schema.yml
Comment (henrystats) [reply models/daos/transactions/ethereum/daos_transactions_ethereum_schema.yml:70]: My God, you're amazing ser!! I'll get on this!

Comment (henrystats) [reply models/daos/transactions/ethereum/daos_transactions_ethereum_schema.yml:70]: Thanksss!

**Comment by henrystats** [comment]
this is ready for review ser @augustog. Sorry for the ping 👀, I've a project that needs this spell that I'm trying to get done by Friday (so wanted to get this merge before Wednesday). 

Thank you very much for all your help @Hosuke

File: models/daos/transactions/gnosis/daos_transactions_gnosis_schema.yml
Comment (Hosuke) [inline models/daos/transactions/gnosis/daos_transactions_gnosis_schema.yml:60]: Do you want to test the gnosis model as well?

Comment (henrystats) [reply models/daos/transactions/gnosis/daos_transactions_gnosis_schema.yml:60]: there's currently an issue with gnosis data where there are some missing transactions, I could run test queries (I already have them set up and included in the code) and then copy the info from gnosis scan. 

But I can do that within the next half an hour or so so yeah let me get onto it.

Comment (henrystats) [reply models/daos/transactions/gnosis/daos_transactions_gnosis_schema.yml:60]: issue on dune*

Comment (henrystats) [reply models/daos/transactions/gnosis/daos_transactions_gnosis_schema.yml:60]: so just checked and I don't have tests for two dao models on gnosis (a transaction query), will have to check the tx_hashes if they pop up.

File: seeds/daos/daos_addresses_seed.csv
Comment (augustog) [reply seeds/daos/daos_addresses_seed.csv]: @Hosuke holy cow yes what a legend

**Comment by henrystats** [comment]
@Hosuke @augustog gnosis data also included in tests. Good for review now ser!

**Review by augustog — APPROVED** [review]
@Hosuke reviewed this thoroughly and I gave it a second pass.
- Metadata & naming conventions look good ✅ 
- Tests added under a generic test framework ✅ ❤️ 

This looks good to merge to me.

FYI @soispoke do you want to validate anything on this PR before starting on DAO spells on our side? Otherwise it should be good to go.

**Comment by soispoke** [comment]
lgtm, the only thing that I'd like to add is that we should use these addresses as labels (in a follow-up PR): https://github.com/duneanalytics/spellbook/tree/main/models/labels

wdyt @henrystats (doesn't have to be done in a hurry but do tag me if you end up opening a label PR) ?

**Review by soispoke — APPROVED** [review]

**Comment by henrystats** [comment]
Hey @soispoke, happy to do a follow up pr later this week to add them as labels. I love the idea of having them as labels. 

Thank you ser!

**Comment by henrystats** [comment]
and I'll make sure to tag you as well once I get it set up!

**Comment by augustog** [comment]
@henrystats
There's a small issue in the config blocks that was detected downstream (since it's a string literal at this level), the fix for that should be small.

Also, the incremental model seems to be failing on merge, possibly due to duplicates. I'd suggest adding a uniqueness test to all tables and look for joins that could be causing duplicates.

Ping me on discord if you need help (same handle as github).

**Comment by augustog** [comment]
Once that is done feel free to open a new PR :)

**Comment by augustog** [comment]
I know this has been a lot of work, sorry we had to revert :(

**Pull request merged**
